### PR TITLE
Update: easyjson, quicktemplate, statik, qtchan, vlang

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
General updates for my packages.
meguca/hydron will come later, pending a possible go2nix issue.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by srhb** [comment]
@grahamcofborg build easyjson quicktemplate statik qtchan vlang

File: pkgs/applications/networking/browsers/qtchan/default.nix [hunk 1]
```diff
@@ -1,23 +1,16 @@
-{ stdenv, fetchFromGitHub, fetchpatch, qt, makeWrapper }:
+{ stdenv, fetchFromGitHub, qt, makeWrapper }:
 
 stdenv.mkDerivation rec {
   name = "qtchan-${version}";
```
Comment (jonringer) [inline pkgs/applications/networking/browsers/qtchan/default.nix]:
```suggestion
  pname = "qtchan";
```

File: pkgs/development/tools/easyjson/default.nix [hunk 2]
```diff
@@ -2,15 +2,15 @@
 
 buildGoPackage rec {
   name = "easyjson-unstable-${version}";
```
Comment (jonringer) [inline pkgs/development/tools/easyjson/default.nix]:
```suggestion
  pname = "easyjson";
```

File: pkgs/development/tools/easyjson/default.nix [hunk 3]
```diff
@@ -2,15 +2,15 @@
...
-  version = "2019-02-21";
+  version = "2019-06-26";
```
Comment (jonringer) [inline pkgs/development/tools/easyjson/default.nix]:
```suggestion
  version = "unstable-2019-06-26";
```

File: pkgs/development/tools/quicktemplate/default.nix [hunk 4]
```diff
@@ -2,15 +2,15 @@
 
 buildGoPackage rec {
   name = "quicktemplate-unstable-${version}";
```
Comment (jonringer) [inline pkgs/development/tools/quicktemplate/default.nix]:
```suggestion
  pname = "quicktemplate";
```

File: pkgs/development/tools/quicktemplate/default.nix [hunk 5]
```diff
@@ -2,15 +2,15 @@
...
-  version = "2019-01-31";
+  version = "2019-07-08";
```
Comment (jonringer) [inline pkgs/development/tools/quicktemplate/default.nix]:
```suggestion
  version = "unstable-2019-07-08";
```

File: pkgs/development/tools/statik/default.nix [hunk 6]
```diff
@@ -2,15 +2,15 @@
 
 buildGoPackage rec {
   name = "statik-unstable-${version}";
```
Comment (jonringer) [inline pkgs/development/tools/statik/default.nix]:
```suggestion
  pname = "statik";
```

File: pkgs/development/tools/statik/default.nix [hunk 7]
```diff
@@ -2,15 +2,15 @@
...
-  version = "2018-11-28";
+  version = "2019-07-31";
```
Comment (jonringer) [inline pkgs/development/tools/statik/default.nix]:
```suggestion
  version = "unstable-2019-07-31";
```

**Review by jonringer — COMMENTED** [review]
pname (I believe) is preferred over name, also, "unstable" shouldn't be part of the package name.

**Comment by worldofpeace** [comment]
I'm seeing a lot of these have `deps.nix`. None of them needed to be updated?

**Review by worldofpeace — CHANGES_REQUESTED** [review]
`qtchan` is packaged improperly for a qt application, and needs updates for `wrapQtAppsHook`.

See the updated manual
* https://hydra.nixos.org/build/96804884/download/1/nixpkgs/manual.html#sec-language-qt

Migration looks like
* call the package with `libsForQt5.callPackage`
* don't reference the `qt` package set in the package
See https://github.com/NixOS/nixpkgs/issues/65399#issuecomment-520915488
* use `mkDerivation` instead of `stdenv.mkDerivation`
* drop custom wrapper in `preFixup`

Note that using qt5's `mkDerivation` automatically includes `wrapQtAppsHook`.

**Comment by Chiiruno** [comment]
>I'm seeing a lot of these have deps.nix. None of them needed to be updated?

go2nix already checks if there are updates needed for deps.nix, there were none needed in this case.

**Comment by Chiiruno** [comment]
That should take care of the issues.

**Review by worldofpeace — APPROVED** [review]
LGTM, executed all binaries. Checked `qtchan` is wrapped.

**Pull request merged**